### PR TITLE
More cli integration tests

### PR DESCRIPTION
**Pull request by therealssj**
- Add test for `verifyAddress`, `listAddresses`, `addressBalance`, `walletBalance`, `decodeRawTransaction`, `addressGen`, `listWallets`, `walletOutputs `, `addressOutputs`
- Move test data from fixtures and golden to testdata folder
- Update `integration-test-stable.sh`; added test selection and update testdata flags
- Update `integration-test-live.sh`; added test selection 

The stable and live integration scripts can now take argument `-t` to select which integration test to run i.e, `cli` or `gui`

The `-u` flag cen be used with `integration-test-stable.sh` to update the golden files

### CODE REVIEW

File: src/api/cli/integration/integration_test.go [hunk 1]
```diff
@@ -63,11 +66,93 @@ func TestGenerateAddresses(t *testing.T) {
...
 }
 
+func TestVerifyAddress(t *testing.T) {
+	if !doLiveOrStable(t) {
+		return
+	}
+
+	output, err := exec.Command(binaryPath, "verifyAddress", "2Kg3eRXUhY6hrDZvNGB99DKahtrPDQ1W9vN").CombinedOutput()
+	require.NoError(t, err)
+	require.Empty(t, output)
```
Comment (gz-c) [inline src/api/cli/integration/integration_test.go]: Test an invalid address
Test a bitcoin address (should be invalid)

File: src/api/cli/integration/integration_test.go [hunk 2]
```diff
@@ -63,11 +66,93 @@ func TestGenerateAddresses(t *testing.T) {
...
+	golden := filepath.Join("testdata", "walletBalance.golden")
+	if *update {
+		writeJSON(t, golden, wltBalance)
+	}
+
+	var expect cli.BalanceResult
+	loadJSON(t, golden, &expect)
+	require.Equal(t, expect, wltBalance)
+}
+
```
Comment (gz-c) [inline src/api/cli/integration/integration_test.go]: For live tests, will have to specify wallet file through an env var, and check that the exit status was fine and the output is parseable

**Comment by gz-c** [comment]
Needs to be `make format` ed

File: src/api/cli/integration/integration_test.go (hunk 2 above)
Comment (therealssj) [reply src/api/cli/integration/integration_test.go]: We already set a wallet in `TestMain`. I have gone ahead and used that only for doing live tests.

https://github.com/therealssj/skycoin/blob/9d1be7c5308546beb7eb024e2006333ba3f1575f/src/api/cli/integration/integration_test.go#L511

File: src/api/cli/integration/integration_test.go [hunk 3]
```diff
@@ -164,6 +180,18 @@ func TestStableAddressBalance(t *testing.T) {
...
+func TestLiveAddressBalance(t *testing.T) {
+	if !doStable(t) {
+		return
+	}
+
+	output, err := exec.Command(binaryPath, "addressBalance", "2kvLEyXwAYvHfJuFCkjnYNRTUfHPyWgVwKt").CombinedOutput()
+	require.NoError(t, err)
+
+	var addrBalance cli.BalanceResult
+	require.NoError(t, json.NewDecoder(bytes.NewReader(output)).Decode(&addrBalance))
```
Comment (gz-c) [inline src/api/cli/integration/integration_test.go]: Always write this as separate steps like

```go
err = json.NewDecoder(bytes.NewReader(output)).Decode(&addrBalance)
require.NoError(t, err)
```

File: src/api/cli/integration/integration_test.go [hunk 4]
```diff
@@ -201,7 +201,7 @@ func TestStableWalletBalance(t *testing.T) {
 	require.NoError(t, err)
 
 	var wltBalance cli.BalanceResult
-	json.Unmarshal(output, &wltBalance)
+	require.NoError(t, json.NewDecoder(bytes.NewReader(output)).Decode(&wltBalance))
```
Comment (gz-c) [inline src/api/cli/integration/integration_test.go]: needs to be on separate lines

**Comment by therealssj** [comment]
Will do rest of the tests related to creating transactions in a separate PR.

**Comment by kittycash-manager** [comment]
Add test cases for options of commands.

**Comment by iketheadore** [comment]
Add test cases for command options

**Pull request merged**